azure-iot-device/azure/iot/device/common/error_map_paho.py

```diff
@@ -0,0 +1,110 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from . import errors
+from . import errors_paho
+import paho.mqtt.client as mqtt
+
+# mapping of Paho conack rc codes to Error object classes
+paho_conack_rc_to_error = {
+    mqtt.CONNACK_REFUSED_PROTOCOL_VERSION: errors_paho.PahoConnectionRefusedProtocolVersionError,
```

Comment:
I'm not sure I agree with the premise of what you're trying to do here.

Why would we 1:1 translate paho errors into erros we define in our SDK? Why turn all Paho errors into a generic (or a small number of more generic) SDK related errors, while preserving the error history to it's initial cause? (i.e. error chaining)